control/tests/margin_test.py

```diff
@@ -11,6 +11,22 @@
 from control.statesp import StateSpace
 from control.margins import *
 
+def assert_array_almost_equal(x, y, ndigit=4):
+
+    x = np.array(x)
+    y = np.array(y)
+    try:
+        if np.isfinite(x).any() and \
+           np.equal(np.isfinite(x), np.isfinite(y)).all() and \
+           np.equal(np.isnan(x), np.isnan(y)).all():
+            np.testing.assert_array_almost_equal(
+                x[np.isfinite(x)], y[np.isfinite(y)], ndigit)
```

Comment:
this would, if I read correctly, consider np.inf and -np.inf to be equal, as long as one element of the arrays was finite.  Is that the intent?